doc/ssdmond/ssdmond-hld.md

```diff
@@ -0,0 +1,176 @@
+
+The `ssdutil` utility assumes that the SSD drive is  `/dev/sda` whereas the drive letter could be any label based on the nuber of SSDs. We leverage the `lsblk` command to get  list of all the SSDs in the device. For instance, if there were three disks labeled 'sda', 'sdb', and 'sdc' on a device, the output of the `lsblk -d -o name,type` command would typically look something like this:
+
+NAME TYPE
+sda  disk
+sdb  disk
+sdc  disk
+
+This output lists the names of the disks (sda, sdb, and sdc) and their types (which are "disk" in this case, indicating that they are block storage devices). This output format makes it easy to identify the disks on any system. We then leverage the following proposed StateDB schema to store and stream information about each of these disks.
+
+## **3. StateDB Schema**
+```
+; Defines information for each SSD in a device
+
+key                 = SSD_INFO|<ssd_name>               ; This key is for information that does not change for the lifetime of the SSD - SSD_INFO|SDX
+
+; field             = value
+
+temperature_celsius = STRING                            ; Describes the operating temperature of the SSD in Celsius                             (Priority 0, Dynamic)
+io_reads            = INT                               ; Describes the total number of reads completed successfully from the SSD               (Priority 0, Dynamic)
+io_writes           = INT                               ; Describes the total number of writes completed on the SSD                             (Priority 0, Dynamic)
+reserved_blocks      = INT                               ; Describes the reserved blocks count of the SSD                                        (Priority 0, Dynamic)
+device_model        = STRING                            ; Describes the Vendor information of the SSD                                           (Priority 1, Static)
+serial              = STRING                            ; Describes the Serial number of the SSD                                                (Priority 1, Static)
+firmware            = STRING                            ; Describes the Firmware version of the SSD                                             (Priority 1, Static)
+health              = STRING                            ; Describes the overall health of the SSD as a % value based on several SMART attrs     (Priority 1, Dynamic)
+```
+
+Example: For an SSD with name 'sda', the STATE_DB entry would be:
+
+```
+127.0.0.1:6379[6]> KEYS SSD_INFO|*
+1) "SSD_INFO|sdb"
+2) "SSD_INFO|sda"
+127.0.0.1:6379[6]> HGETALL SSD_INFO|SDA
+ 1) "temperature"
+ 2) "30C"
+ 3) "io_reads"
+ 4) "49527"
+ 5) "io_writes"
+ 6) "238309"
+ 7) "reserved_blocks"
+ 8) "0"
+ 9) "device_model"
+10) "InnoDisk Corp. - mSATA 3IE3"
+11) "health"
+12) "92"
+13) "serial"
+14) "BCA11712190600251"
+15) "firmware"
+16) "S16425cG"
+127.0.0.1:6379[6]> 
+```
+
+## Future Work
+
+1. Support for eMMC storage
+2. Support for more SSD vendors
```

Comment:
I think we should add a line about `Support for nvme storage` since it's likely to supersede ssd over time.

In turn that made me wonder if we should be calling this hld `Storage Monitoring` instead of `SSD Monitoring`.
`Storage` has the advantage of being technology independent and if we end up monitoring `eMMC`, `SSD`, `nvme`, ... that might actually be a bit more future proof.